app/controllers/users_controller.rb

```diff
@@ -0,0 +1,54 @@
+class UsersController < ApplicationController
+
+  get '/signup' do
+    if !session[:user_id]
+      erb :'/users/signup'
+    else
+      redirect '/tweets'
+    end
+  end
+
+  post '/signup' do
+    if params[:username] == "" || params[:email] == "" || params[:password] == ""
+      redirect '/signup'
+    else
+      @user = User.create(params)
+      session[:user_id] = @user.id
+      # binding.pry
+      redirect '/tweets'
+    end
+  end
+
+  get '/login' do
+    if !session[:user_id]
+      erb :'/users/login'
+    else
+      redirect '/tweets'
+    end
+  end
+
+  post '/login' do
+    user = User.find_by(username: params[:username])
```

Comment:
does this work if there are two users that have the same username? Have you set up validations in your user model to protect against duplicate usernames?